bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/HueBridge.java

```diff
@@ -1129,4 +971,133 @@ private String getRelativeURL(String path) {
+            logger.trace("Hue response: status = {}, content = '{}'", httpStatus, content);
+            return new HueResult(content, httpStatus);
+        } catch (ExecutionException e) {
+            String message = e.getMessage();
+            if (e.getCause() instanceof SSLHandshakeException) {
+                logger.debug("SSLHandshakeException occurred during execution: {}", message, e);
+                throw new ConfigurationException(TEXT_OFFLINE_CONFIGURATION_ERROR_INVALID_SSL_CERIFICATE, e.getCause());
+            } else {
+                logger.debug("ExecutionException occurred during execution: {}", message, e);
+                throw new CommunicationException(message == null ? TEXT_OFFLINE_COMMUNICATION_ERROR : message,
+                        e.getCause());
+            }
+        } catch (TimeoutException e) {
+            String message = e.getMessage();
+            logger.debug("TimeoutException occurred during execution: {}", message, e);
+            throw new CommunicationException(message == null ? TEXT_OFFLINE_COMMUNICATION_ERROR : message);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            String message = e.getMessage();
+            logger.debug("InterruptedException occurred during execution: {}", message, e);
+            throw new CommunicationException(message == null ? TEXT_OFFLINE_COMMUNICATION_ERROR : message);
+        }
+    }
+
+    private CompletableFuture<HueResult> putAsync(String address, String body, long delay) {
+        AsyncPutParameters asyncPutParameters = new AsyncPutParameters(address, body, delay);
+        synchronized (commandsQueue) {
+            if (commandsQueue.isEmpty()) {
+                commandsQueue.offer(asyncPutParameters);
+                Future<?> localJob = job;
+                if (localJob == null || localJob.isDone()) {
+                    job = scheduler.submit(this::executeCommands);
+                }
+            } else {
+                commandsQueue.offer(asyncPutParameters);
+            }
+        }
+        return asyncPutParameters.future;
+    }
+
+    private void executeCommands() {
+        while (true) {
+            try {
+                long delayTime = 0;
+                synchronized (commandsQueue) {
+                    AsyncPutParameters payloadCallbackPair = commandsQueue.poll();
+                    if (payloadCallbackPair != null) {
+                        logger.debug("Async sending put to address: {} delay: {} body: {}", payloadCallbackPair.address,
+                                payloadCallbackPair.delay, payloadCallbackPair.body);
+                        try {
+                            HueResult result = doNetwork(payloadCallbackPair.address, HttpMethod.PUT,
+                                    payloadCallbackPair.body);
+                            payloadCallbackPair.future.complete(result);
+                        } catch (CommunicationException e) {
```

Comment:
Are you not catching also ConfigurationException ?

Reply:
I've checked that it's caught in current code.